paasta_tools/docker_wrapper.py

```diff
@@ -134,6 +139,51 @@ def can_add_mac_address(args):
     return True
 
 
+def can_add_ip_address(args):
+    # Can only specify an IP address for user defined networks
+    if is_network_host(args) or not is_run(args):
+        return False
+
+    for index, arg in enumerate(args):
+        # Check for --ip
+        if arg.startswith('--ip'):
+            return False
+
+    return True
+
+
+def get_ip_for_container():
+    # This would look at the currently outstanding IP addresses
+    # Something like but not necessarily =>
+    current_ip_addresses = subprocess.check_output((
+        "for i in $(sudo docker network ls | cut -d ' ' -f 1 | tail -n +2); "
+        "do "
+            "sudo docker network inspect $i; "
+        "done "
+            "| jq -r '.[].Containers | to_entries | .[].value.IPv4Address' "
+            "| sed 's/\/.*$//g'"
+    ), shell=True).split()
+    # Then try to get one in the subnet of the docker
+    # Ideally get this from config etc
+    subnet = ip_network(u'169.254.16.0/24', False)
+    hosts = list(subnet.hosts())
+    new_ip = random.choice(hosts)
```

Comment:
You can just use sets and subtract: `hosts - current_ip_addresses` and then random choice from the leftovers